denops_std/buffer/decoration.ts

```diff
@@ -94,10 +96,12 @@ async function vimDecorate(
 async function vimUndecorate(
   denops: Denops,
   bufnr: number,
+  start: number,
+  end: number,
 ): Promise<void> {
-  const propList = await vimFn.prop_list(denops, 1, {
+  const propList = await vimFn.prop_list(denops, start + 1, {
     bufnr,
-    end_lnum: -1,
+    end_lnum: end,
```

Comment:
:question:

Not suer if it works properly when the `end` is NOT `-1`. Could you confirm it?

Reply:
From help.
```
prop_list({lnum} [, {props}])                           *prop_list()*
                Returns a List with all the text properties in line {lnum}.

                The following optional items are supported in {props}:
                   bufnr        use this buffer instead of the current buffer
                   end_lnum     return text properties in all the lines
                                between {lnum} and {end_lnum} (inclusive).
                                A negative value is used as an offset from the
                                last buffer line; -1 refers to the last buffer
                                line.
```
In vim, 1-index and inclusive. In neovim, 0-index and exclusive.
So, no problem.